src/module.c

```diff
@@ -482,20 +482,21 @@ JL_DLLEXPORT jl_value_t *jl_get_global(jl_module_t *m, jl_sym_t *var)
 JL_DLLEXPORT void jl_set_global(jl_module_t *m, jl_sym_t *var, jl_value_t *val)
 {
     jl_binding_t *bp = jl_get_binding_wr(m, var, 1);
-    if (!bp->constp) {
-        bp->value = val;
-        jl_gc_wb(m, val);
-    }
+    assert(!bp->constp && "Can't modify const; consider jl_checked_assignment");
+    bp->value = val;
+    jl_gc_wb(m, val);
 }
 
-JL_DLLEXPORT void jl_set_const(jl_module_t *m, jl_sym_t *var, jl_value_t *val)
+JL_DLLEXPORT void jl_define_const(jl_module_t *m, jl_sym_t *var, jl_value_t *val)
 {
-    jl_binding_t *bp = jl_get_binding_wr(m, var, 1);
-    if (!bp->constp) {
-        bp->value = val;
-        bp->constp = 1;
-        jl_gc_wb(m, val);
-    }
+    // Assert that it doesn't already exist.
+    jl_binding_t *bp = jl_get_binding(m, var);
```

Comment:
This won't work, since it will cause the binding to be resolved (possibly to something visible via `using`), when we want to create a new variable owned by this module. The binding also doesn't have to be totally absent; other operations might have added a binding object to the table without setting a value for the variable.

Reply:
Ah, okay, thanks. Can i just check if the `value` is `NULL`, then? Is a `NULL` `jl_binding_t.value` different than a variable whose value is `nothing`? I'm not that familiar with the C api still, sorry.

Reply:
(Also, sorry i haven't been testing these commits before pushing.. I'm having trouble getting my local julia to build for some reason. i might need to blow it away and start over.)